main.c

```diff
@@ -730,21 +730,16 @@ static void run(void)
 				if (!tns_load(&tns, tns.initnext, false, true))
 					remove_file(tns.initnext, false);
 			} else {
-				xfd = ConnectionNumber(win.env.dpy);
-				FD_ZERO(&fds);
-				FD_SET(xfd, &fds);
-				if (info.fd != -1) {
-					FD_SET(info.fd, &fds);
-					xfd = MAX(xfd, info.fd);
-				}
-				if (arl.fd != -1) {
-					FD_SET(arl.fd, &fds);
-					xfd = MAX(xfd, arl.fd);
-				}
-				select(xfd + 1, &fds, 0, 0, to_set ? &timeout : NULL);
-				if (info.fd != -1 && FD_ISSET(info.fd, &fds))
+				pfd[FD_X].fd = ConnectionNumber(win.env.dpy);
+				pfd[FD_INFO].fd = info.fd;
+				pfd[FD_ARL].fd = arl.fd;
+				pfd[FD_X].events = pfd[FD_INFO].events = pfd[FD_ARL].events = POLLIN;
+
+				if (poll(pfd, ARRLEN(pfd), to_set ? TV_TO_MS(&timeout) : -1) < 0)
+					continue;
```

Comment:
So just ignore if there are any errors? I think it's reasonable to not just exit/kill the program but, what about at least sending `perror()`?

Reply:
Righttt sxiv uses SIGCHLD and SIGPIPE (not sure what this one does).. Fallbacking to the default approach to wait for a complaint if any.